cleverhans/dataset.py

```diff
@@ -0,0 +1,144 @@
+"""Dataset class for CleverHans
+
+"""
+
+from __future__ import absolute_import
+from __future__ import division
+from __future__ import print_function
+
+from cleverhans.utils_mnist import data_mnist
+
+keras = None  # Only load keras if user tries to use a dataset that requires it
+
+
+class Dataset(object):
+    """Abstract base class representing a dataset.
+    """
+
+    # The number of classes in the dataset. Should be specified by subclasses.
+    NB_CLASSES = None
+
+    def __init__(self, **kwargs):
+        if "self" in kwargs:
+            del kwargs[self]
+        self.kwargs = kwargs
+
+    def get_factory(self):
+        """Returns a picklable callable that recreates the dataset.
+        """
+
+        if hasattr(self, 'kwargs'):
+            kwargs = self.kwargs
+        else:
+            kwargs = {}
+
+        return Factory(type(self), kwargs)
+
+
+class MNIST(Dataset):
+    """The MNIST dataset"""
+
+    NB_CLASSES = 10
+
+    def __init__(self, train_start=0, train_end=60000, test_start=0,
+                 test_end=10000, center=False):
+        super(MNIST, self).__init__(**locals())
+        self.kwargs = locals()
```

Comment:
This statement is not longer necessary since you moved the logic to the constructor.

Reply:
thanks, done